services/src/main/java/org/keycloak/storage/RoleStorageManager.java

```diff
@@ -69,6 +74,14 @@ public static RoleStorageProvider getStorageProvider(KeycloakSession session, Re
 
 
     public static <T> Stream<RoleStorageProviderModel> getStorageProviders(RealmModel realm, KeycloakSession session, Class<T> type) {
+        List<ProviderFactory> factories = session.getKeycloakSessionFactory()
+                .getProviderFactories(RoleStorageProvider.class);
+
+        // there is no need iterate over the database if there is no role storage provider
+        if (factories.isEmpty()) {
+            return Stream.empty();
+        }
+
```

Comment:
@mhajas, I believe this is correct but would like you to check

Reply:
This looks good to me, I like it. However, we should probably add it also to [AbstractStorageManager](https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/storage/AbstractStorageManager.java#L93), it should be used everywhere in future, so I think this is the correct place to add it.

Reply:
@mhajas I've added the check at that method. Is that what you are looking for?

Reply:
@pedroigor yes, exactly. Thanks!